### PR TITLE
246 remove cxf

### PR DESCRIPTION
**Pull request by erdlet**
Removes CXF module + workarounds in the Core module.

Also...

- ... adapts the documentation
- ... removes TomEE from TCK / Testsuite profiles
- ... comments out OpenLiberty profile / adapts TCK Archive creation for later usage

In Jenkins, I disabled the TCK job for openliberty, so it won't fail because of the missing CXF dependency (it was red all the time, so I think that won't hurt us).

### CODE REVIEW

**Review by lefloh — APPROVED** [review]

File: tck/pom.xml [hunk 1]
```diff
@@ -266,53 +260,53 @@
...
-            </build>
-            <dependencies>
-                <dependency>
-                    <groupId>io.openliberty.arquillian</groupId>
-                    <artifactId>arquillian-liberty-managed-jakarta</artifactId>
-                    <version>2.0.0-M1</version>
-                    <scope>test</scope>
-                </dependency>
-            </dependencies>
+<!--         <profile> -->
```
Comment (ivargrimstad) [inline tck/pom.xml:263]: Uncomment

File: testsuite/pom.xml [hunk 2]
```diff
@@ -335,79 +335,42 @@
...
-                            <systemProperties>
-                                <arquillian.launch>liberty</arquillian.launch>
-                                <testsuite.profile>testsuite-liberty</testsuite.profile>
-                            </systemProperties>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
+<!--         <profile> -->
```
Comment (ivargrimstad) [inline testsuite/pom.xml:339]: Uncomment

**Review by ivargrimstad — CHANGES_REQUESTED** [review]
Latest versions of OpenLiberty 21.* comes with RESTEasy, so we should keep those tests in. Even if the tests fail currently, but I think that is more of a setup issue.

**Comment by erdlet** [comment]
@ivargrimstad Thanks for your review. I've uncomment the Liberty configs and fixed the documentation. Also I re-enabled the TCK job and extended the testsuite job, so we can see if everything works when we use OpenLiberty with Krazo RESTEasy.

**Comment by ivargrimstad** [comment]
Great!

The Jobs are set up since a while back in our Jenkins: https://ci.eclipse.org/krazo/ 

The EE4J GitHub organization is too big to be useful for Travis other than just doing the simple compile and run unit tests. One spin of our TCK exhausted all the resources so it was useless for 24 hours :)

**Review by ivargrimstad — APPROVED** [review]

**Comment by erdlet** [comment]
> The Jobs are set up since a while back in our Jenkins: https://ci.eclipse.org/krazo/

Yeah you demonstrated ist during the last call :) I added the config there, hopefully correct.

Maybe it makes sense to clean the Travis config in a new PR 🤔

**Review by chkal — APPROVED** [review]
Wow! We had more CXF workarounds in our codebase than I was aware of. Happy to see them being removed. :+1:

**Review by Daniel-Dos — APPROVED** [review]

**Review by gtudan — APPROVED** [review]

**Comment by erdlet** [comment]
I'll merge this now, so we can see if something is broken. Thanks for your reviews :beers:

**Pull request merged**